proton-j/src/main/java/org/apache/qpid/proton/codec/CompositeReadableBuffer.java

```diff
@@ -834,22 +834,39 @@ public boolean equals(Object other) {
             return false;
         }
 
-        ReadableBuffer buffer = (ReadableBuffer)other;
-        if (this.remaining() != buffer.remaining()) {
+        ReadableBuffer buffer = (ReadableBuffer) other;
+        final int remaining = remaining();
+        if (remaining != buffer.remaining()) {
             return false;
         }
+        if (hasArray()) {
+            return equals(currentArray, position, remaining, buffer);
```

Comment:
This bit is a little broken as it uses position without account for any array offset in the current buffer. Its position in the buffer may not be the same as its position within the array, e.g. if it were previously sliced from a larger buffer. Calling equals with slices from two equal buffers with different current positions will fail.

Reply:
So ti should take into account the arrayOffset and do not rely on position?

Reply:
For the single array case, when you have a slice that you've created from a CRB that is not starting from zero then the current offset will be non-zero but position would be so the equals check should be starting at current offset and moving forward however many elements the limit allows for.  

The position and the offset work together to control where zero is in the buffer, you can do a get for some index in the buffer and the position controls how far back you can rewind the offset until you've hit "zero" which might not be zero in the array.